test/e2e/aks.go

```diff
@@ -266,7 +267,7 @@ func GetWorkingAKSKubernetesVersion(ctx context.Context, subscriptionID, locatio
 	if err != nil {
 		return "", errors.Wrap(err, "failed to create an Authorizer")
 	}
-	containerServiceClient := containerservice.NewContainerServicesClient(subscriptionID)
+	containerServiceClient := containerservice2020.NewContainerServicesClient(subscriptionID)
 	containerServiceClient.Authorizer = authorizer
 	result, err := containerServiceClient.ListOrchestrators(ctx, location, ManagedClustersResourceType)
```

Comment:
I should have linked to [AgentPoolsClient.GetAvailableAgentPoolVersions](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v63.4.0+incompatible/services/containerservice/mgmt/2022-03-01/containerservice#AgentPoolsClient.GetAvailableAgentPoolVersions)

Reply:
The main difference I see with that function is that it requires an existing cluster which our existing function doesn't, since we're introspecting versions before creating the cluster

I've asked some folks on the AKS team to help

Reply:
Actually, neither of those are going to work because they need a running cluster in order to lookup the possible versions for that cluster. In our scenario we need the version number in order to _create_ the cluster.

Reply:
Well, I checked with the latest Python client and it downgrades the api version it calls to get the list of kubernetes versions available.

```
urllib3.connectionpool: https://management.azure.com:443 "GET /subscriptions/REDACTED/providers/Microsoft.ContainerService/locations/eastus/orchestrators?api-version=2019-04-01&resource-type=managedClusters HTTP/1.1" 200 None
```

```
$> az version                                   
{
  "azure-cli": "2.42.0",
  "azure-cli-core": "2.42.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "aks-preview": "0.5.117"
  }
}
```

@CecileRobertMichon I'm going to need some help determining if there's an updated equivalent of the Orchestrators API but so far I can't seem to find one to list the Kubernetes API versions.

Reply:
@mtougeron I checked with AKS folks and currently there is no replacement :( They have plans to add a new API to list versions but for now looks like we're stuck using the old SDK.

Reply:
Thanks for checking, I appreciate it.